src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -260,32 +277,32 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             options &= protocolMask;
             Debug.Assert(options != protocolMask, "All protocols are disabled");
 
-            var noSsl2 = (options & libssl.Options.SSL_OP_NO_SSLv2) != 0;
-            var noSsl3 = (options & libssl.Options.SSL_OP_NO_SSLv3) != 0;
-            var noTls10 = (options & libssl.Options.SSL_OP_NO_TLSv1) != 0;
-            var noTls11 = (options & libssl.Options.SSL_OP_NO_TLSv1_1) != 0;
-            var noTls12 = (options & libssl.Options.SSL_OP_NO_TLSv1_2) != 0;
+            bool noSsl2 = (options & libssl.Options.SSL_OP_NO_SSLv2) != 0;
+            bool noSsl3 = (options & libssl.Options.SSL_OP_NO_SSLv3) != 0;
+            bool noTls10 = (options & libssl.Options.SSL_OP_NO_TLSv1) != 0;
+            bool noTls11 = (options & libssl.Options.SSL_OP_NO_TLSv1_1) != 0;
+            bool noTls12 = (options & libssl.Options.SSL_OP_NO_TLSv1_2) != 0;
 
-            var method = IntPtr.Zero;
+            IntPtr method = IntPtr.Zero;
             if (noSsl2 && noSsl3 && noTls11 && noTls12)
             {
-                method = isServer ? libssl.TLSv1_server_method() : libssl.TLSv1_client_method();
+                method = libssl.TLSv1_method();
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls12)
             {
-                method = isServer ? libssl.TLSv1_1_server_method() : libssl.TLSv1_1_client_method();
+                method = libssl.TLSv1_1_method();
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls11)
             {
-                method = isServer ? libssl.TLSv1_2_server_method() : libssl.TLSv1_2_client_method();
+                method = libssl.TLSv1_2_method();
             }
             else if (noSsl2 && noTls10 && noTls11 && noTls12)
             {
-                method = isServer ? libssl.SSLv3_server_method() : libssl.SSLv3_client_method();
+                method = libssl.SSLv3_method();
             }
             else
             {
-                method = isServer ? libssl.SSLv23_server_method() : libssl.SSLv23_client_method();
+                method = libssl.SSLv23_method();
```

Comment:
Do these functions return values that never change for the lifetime of the process?  If so, we might consider caching them into statics so that we can avoid these managed/native calls every time GetSslMethod is called.  Unless GetSslMethod isn't called frequently?